src/fastgltf_tools.hpp

```diff
@@ -0,0 +1,303 @@
+
+		auto& valuesView = asset.bufferViews[accessor.sparse->bufferViewValues];
+		auto* valuesBytes = adapter(asset.buffers[valuesView.bufferIndex])
+				+ valuesView.byteOffset + accessor.sparse->byteOffsetValues;
+		// "The index of the bufferView with sparse values. The referenced buffer view MUST NOT
+		// have its target or byteStride properties defined."
+		auto valueStride = getElementByteSize(accessor.type, accessor.componentType);
+
+		for (std::size_t i = 0; i < accessor.sparse->count; ++i) {
+			auto idx = internal::getAccessorElementAt<std::uint32_t>(accessor.sparse->indexComponentType,
+					indicesBytes + indexStride * i);
+			func(internal::getAccessorElementAt<ElementType>(accessor.componentType,
+					valuesBytes + valueStride * idx));
+		}
+	}
+	else {
+		auto& view = asset.bufferViews[*accessor.bufferViewIndex];
+		auto stride = view.byteStride ? *view.byteStride : getElementByteSize(accessor.type, accessor.componentType);
+
+		auto* bytes = adapter(asset.buffers[view.bufferIndex]);
+		bytes += view.byteOffset + accessor.byteOffset;
+
+		for (std::size_t i = 0; i < accessor.count; ++i) {
+			func(internal::getAccessorElementAt<ElementType>(accessor.componentType, bytes + i * stride));
+		}
+	}
+
+}
+
+template <typename ElementType, std::size_t TargetStride = sizeof(ElementType),
+		 typename BufferDataAdapter = DefaultBufferDataAdapter>
+void fillTargetFromAccessor(const Asset& asset, const Accessor& accessor, void* dest,
+		const BufferDataAdapter& adapter = {}) {
+	if (accessor.type != ElementTraits<ElementType>::type) {
+		return;
+	}
+
+	auto* dstBytes = reinterpret_cast<std::byte*>(dest);
+
+	if (accessor.sparse) {
+		auto& indicesView = asset.bufferViews[accessor.sparse->bufferViewIndices];
+		auto* indicesBytes = adapter(asset.buffers[indicesView.bufferIndex])
+				+ indicesView.byteOffset + accessor.sparse->byteOffsetIndices;
+		auto indexStride = getElementByteSize(AccessorType::Scalar, accessor.sparse->indexComponentType);
+
+		auto& valuesView = asset.bufferViews[accessor.sparse->bufferViewValues];
+		auto* valuesBytes = adapter(asset.buffers[valuesView.bufferIndex])
+				+ valuesView.byteOffset + accessor.sparse->byteOffsetValues;
+		// "The index of the bufferView with sparse values. The referenced buffer view MUST NOT
+		// have its target or byteStride properties defined."
+		auto srcStride = getElementByteSize(accessor.type, accessor.componentType);
+
+		for (std::size_t i = 0; i < accessor.sparse->count; ++i) {
+			auto idx = internal::getAccessorElementAt<std::uint32_t>(accessor.sparse->indexComponentType,
+					indicesBytes + indexStride * i);
+			auto* pDest = reinterpret_cast<ElementType*>(dstBytes + TargetStride * i);
+
+			*pDest = internal::getAccessorElementAt<ElementType>(accessor.componentType,
+					valuesBytes + srcStride * idx);
+		}
+	}
+	else {
+		// FIXME: Assuming that I have a buffer view index, but what do I do if I don't?
```

Comment:
Here's what you do:
> 5.1.1. accessor.bufferView
The index of the buffer view. When undefined, the accessor MUST be initialized with zeros; sparse property or extensions MAY override zeros with actual values.

Same applies to the `getAccessorElement` function which also doesn't check if a value is given.

Reply:
Resolved in [472b1db](https://github.com/spnda/fastgltf/pull/20/commits/472b1dbdffc920d8731b29e0cb1e87d62126b1a1)